### PR TITLE
ThreadX Utils

### PR DESCRIPTION
**Pull request by caiodasilva2005**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: threadX/src/u_debug.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+const char* hal_status_toString(HAL_StatusTypeDef status) {
+    switch(status) {
+        case HAL_OK: return "HAL_OK";
+        case HAL_ERROR: return "HAL_ERROR";
+        case HAL_BUSY: return "HAL_BUSY";
+        case HAL_TIMEOUT: return "HAL_TIMEOUT";
+        default: return "UNKNOWN_STATUS";
+    }
+}
```
Comment (bjackson312006) [inline threadX/src/u_debug.c:54]: There's probably gonna have to be a NetX-specific directory in embedded-base (assuming the Ethernet stuff works), so `nx_status_toString()` could be added in there (`netX/src/unx_debuc.g` maybe?)

**Review by bjackson312006 — CHANGES_REQUESTED** [review]
